src/Rings/groebner.jl

```diff
@@ -1362,3 +1365,74 @@ function _extend_mon_order(ordering::MonomialOrdering,
   m_hom[2:end, 2:end] = m
   return matrix_ordering(homogenized_ring, m_hom)
 end
+
+function _mod_rand_prime(I::MPolyIdeal)
+  p = 32771
+  while true
+    p = Hecke.next_prime(p)
+    
+    base_field = GF(p)
+    ModP, _ = PolynomialRing(base_field, "x" => 1:ngens(base_ring(I)))
+    I_mod_p_gens =
+      try
+        [map_coefficients(base_field, f; parent=ModP) for f in gens(I)]
+      catch e
+        # this precise error is thrown if the chosen prime p divides one
+        # of the denominators of the coefficients of the generators of I.
+        # In this case we simply choose the next prime and try again.
+        if e == ErrorException("Unable to coerce") 
+          continue
+        else
+          rethrow(e)
+        end
+      end
+    return ideal(ModP, I_mod_p_gens)
+  end
+end
+
+# check homogeneity w.r.t. some weights
+
+function _is_homogeneous(f::MPolyElem, weights::Vector{Int})
+  all([sum(weights .* e) == sum(weights .* first(exponents(f)))
+       for e in exponents(f)])
+end
+
+function _is_homogeneous(f::MPolyElem)
+  _is_homogeneous(f, ones(Int, ngens(parent(f))))
+end
+  
+
+# compute weights such that F is a homogeneous system w.r.t. these weights
+function _find_weights(F::Vector{P}) where {P <: MPolyElem}
+  nrows = sum((length).(F)) - length(F)
+  ncols = ngens(parent(first(F)))
+  mat_space = MatrixSpace(QQ, nrows, ncols)
+
+  exp_diffs = permutedims(reduce(hcat, [e[i] - e[1] for e in
+                                          (collect).((exponents).(F))
+                                          for i in 2:length(e)]))
+  K = kernel(mat_space(exp_diffs))[2]
+  isempty(K) && return zeros(Int, ncols)
+  # Here we try to find a vector with strictly positive entries in K
+  # this method to find such a vector is taken from
+  # https://mathoverflow.net/questions/363181/intersection-of-a-vector-subspace-with-a-cone
+  Pol = Polyhedron(-K,  zeros(Int, ncols))
+  !is_feasible(Pol) && return zeros(Int, ncols)
+  pos_vec = zeros(Int, ncols)
+  for i in 1:ncols
+    ei = [j == i ? one(QQ) : zero(QQ) for j in 1:ncols]
+    obj_func = ei * K
+    L = LinearProgram(Pol, obj_func)
+    m, v = solve_lp(L)
+    if isnothing(v)
+      Pol_new = intersect(Pol, Polyhedron(ei*K, [1]))
+      L = LinearProgram(Pol_new, obj_func)
+      v = optimal_vertex(L)
+    end
+    pos_vec += K*(v.p)
+  end
+  ret = (Int).(lcm((denominator).(pos_vec)) .* pos_vec)
```

Comment:
Shouldn't we be careful with the possible degrees? Like say degree should be `<2^16` for faster Singular computations?

Reply:
`Singular.std_hilbert` accepts `Int32` weights, is this the only restriction we should enforce on the weights? I have no idea if the speed of the computation is otherwise affected by the size of the weights.